homework/index.js

```diff
@@ -27,18 +12,130 @@
         elem.setAttribute(key, value);
       }
     });
+
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
+  function buildRepositoryContributors(contributors, parent) {
+    createAndAppend('label', parent, { text: 'Contributors', id: 'titleContributor' });
+    contributors.forEach(contributor => {
+      const contributorData = createAndAppend('div', parent, { class: 'contributor' });
+      createAndAppend('img', contributorData, {
+        src: contributor.avatar_url,
+        alt: contributor.login,
+      });
+      contributorData.addEventListener('click', () => {
+        window.open(contributor.html_url, '_blank');
+      });
+      contributorData.addEventListener('mouseover', () => {
+        contributorData.style.cursor = 'pointer';
+      });
+      createAndAppend('span', contributorData, { text: contributor.login, class: 'name' });
+      createAndAppend('span', contributorData, {
+        text: contributor.contributions,
+        class: 'numberOfContribution',
+      });
+    });
+    return parent;
+  }
+  const fetchJSON = url => {
+    const promise = new Promise((resolve, reject) => {
```

Comment:
Hi Mohamd, even though this is a pattern of wrapping callbacks to promises which you probably would use later in the future but as far as this week's homework is concerned, you are to use the browser native `fetch` API. Please find the documentation around it here - https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch

The native API works quite similar to what you are doing here explicitly. Thus, you need not use this function in your assignment. But please be mindful of this statement in the documentation above: `An accurate check for a successful fetch() would include checking that the promise resolved, then checking that the Response.ok property has a value of true.`

This means that even if a promise is resolved, it doesn't mean that the request was successful. It could have failed as well. Which is why, we have an extra check for `response.ok` to make sure that our response was successful. The documentation provides a code snippet about how you can use it. Please refer the same.